### ISSUE TITLE
Facet counts are incorrect when combined with grouping and pinning

### ISSUE THREAD

**Issue by 527bd5** [typesense/typesense#1224]
## Description

The facet counts are incorrect a filter is applied and you combine `group_by` with `pinned_hits`.

## Steps to reproduce

See [this gist](https://gist.github.com/527bd5/97bb7564f720eea7006b8b148a17fbb8).

## Expected Behavior

Detailed in the gist, but I expect to get correct facet counts.

## Actual Behavior

You get the correct facet count - the number of pinned hits that satisfy the filter.

## Metadata

**Typesense Version**: Tried on 0.25.0.rc25, 0.25.2.rc1-arm64 and 0.26.0.rc18.

**OS**: Running ARM Docker build on macOS

**Comment by vamshiaruru32** [comment]
We might be facing a similar issue. When using group by in conjecture with pinned hits, typesense duplicates the search results and sends multiple results with the same group key. For a query like this

```
curl --location --globoff '<url>/collections/product_variants/documents/search?q=*&group_by=productSlug&query_by=searchText0%2CsearchText1%2CsearchText2&group_limit=1&per_page=100&pinned_hits=gid-shopify-productvariant-40595546341511%3A1%2Cgid-shopify-productvariant-40606843207815%3A2&filter_by=productSlug%20%3A%3D%20[%60butterscotch-oversized-top%60%2C%60butterscotch-square-neck-maxi-dress%60]&include_fields=id' \
--header 'x-typesense-api-key: <api_key>' \
--data ''
```
we get a result like this
```
{
    "found": 3,
    "grouped_hits": [
        {
            "group_key": [
                "butterscotch-oversized-top"
            ],
            "hits": [
                {
                    "curated": true,
                    "document": {
                        "id": "gid-shopify-productvariant-40595546341511"
                    },
                    "highlight": {},
                    "highlights": []
                }
            ]
        },
        {
            "group_key": [
                "butterscotch-square-neck-maxi-dress"
            ],
            "hits": [
                {
                    "curated": true,
                    "document": {
                        "id": "gid-shopify-productvariant-40606843207815"
                    },
                    "highlight": {},
                    "highlights": []
                }
            ]
        },
        {
            "found": 5,
            "group_key": [
                "butterscotch-square-neck-maxi-dress"
            ],
            "hits": [
                {
                    "document": {
                        "id": "gid-shopify-productvariant-40606843142279"
                    },
                    "highlight": {},
                    "highlights": []
                }
            ]
        }
    ],
}
```
as you can see, butterscotch-square-neck-maxi-dress is repeated twice - once for the pinned hit and once without the pinned hit. This bug doesn't arise if group by is not present.

**Comment by kishorenc** [comment]
Fix available in `0.26.0.rc65`.

**Comment by jasonbosco** [comment]
Released in v26
